scripts/core/superdesk-api.d.ts

```diff
@@ -1,9 +1,41 @@
+/* tslint:disable */
+
 declare module 'superdesk-api' {
-    export interface IExtension {
-        activate: (superdesk: ISuperdesk) => void;
+    // TYPESCRIPT TYPES
+
+    export interface DeepReadonlyArray<T> extends ReadonlyArray<DeepReadonly<T>> {};
+
+    export type DeepReadonlyObject<T> = {
+        readonly [P in keyof T]: DeepReadonly<T[P]>;
+    }
+
+    export type DeepReadonly<T> =
+        T extends Function
+            ? T
+            : T extends Array<infer U>
+                ? DeepReadonlyArray<U>
+                : DeepReadonlyObject<T>;
+
+    export type Omit<K, V> = Pick<K, Exclude<keyof K, V>>;
+
+
+
+    // EXTENSIONS
+
+    export interface IExtensionActivationResult {
+        contributions?: {
+            editor3?: {
+                annotationInputTabs?: Array<IEditor3AnnotationInputTab>;
+            }
+            pages?: Array<IPage>;
+        }
     }
 
-    export interface IExtensionObject {
+    export type IExtension = DeepReadonly<{
+        activate: (superdesk: ISuperdesk) => Promise<IExtensionActivationResult>;
```

Comment:
I don't see extensions as having a **value**. Such value doesn't represent the entire extension, but only the functions it registered for the core to call.

I'm not sure about other methods. The plan is to allow extensions to communicate with each other, so other functions might be exported for that or it might not.